README.md

```diff
@@ -102,6 +102,25 @@ the secrets stored in temp files and in the Python process environment are gone.
 
     This flag can be useful for when you have secrets that you don't need access to for development. For example API keys for monitoring tools. This flag can be used multiple times.
 
+* `-e, --environment` Specify section (environment) to parse from secret YAML
+
+    This flag specifies which specific environment/section to parse from the secrets YAML file (or string). In addition, it will also enable the usage of a `common` section which will be inherited by other sections/environments. In other words, if your `secrets.yaml` looks something like this:
+
+```yaml
+common:
```

Comment:
I think it would be a nice 12 factor separation of concerns though. The codebase shouldn't need to be aware of the environments the service is running in (imagine an open source service?). The codebase should be environment agnostic and should simply define the ENVs required to run the service.

The environments the service runs in can be determined by the S3 buckets. And regarding a trail, I _think_ S3 buckets supports versioning (some sort of source control basically).